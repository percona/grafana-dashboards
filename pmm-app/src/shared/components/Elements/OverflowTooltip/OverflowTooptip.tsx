import React, { useEffect, useRef, useState } from 'react';

export const OverflowTooltip = ({ children, ...props }) => {
  const [title, setTitle] = useState('');
  const ref = useRef<HTMLSpanElement>(null);

  useEffect(() => {
    if (ref.current && ref.current.clientWidth < ref.current.scrollWidth) {
      setTitle(children);
    }
  }, [ref]);

  return (
    <span ref={ref} title={title} {...props}>{children}</span>
  );
};
