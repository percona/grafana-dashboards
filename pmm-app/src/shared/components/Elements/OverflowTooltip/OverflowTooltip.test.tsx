import React from 'react';
import { shallow } from 'enzyme';
import { OverflowTooltip } from './OverflowTooptip';

xdescribe('OverflowTooltip test', () => {
  it('OverflowTooltip renders correctly with children', () => {
    const tree = shallow(<OverflowTooltip>Test label</OverflowTooltip>);

    expect(tree).toMatchSnapshot();
    tree.unmount();
  });
});
