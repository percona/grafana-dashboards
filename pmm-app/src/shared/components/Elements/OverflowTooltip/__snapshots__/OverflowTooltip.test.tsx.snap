// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OverflowTooltip test OverflowTooltip renders correctly with children 1`] = `
<span
  title=""
>
  Test label
</span>
`;
